Yosemite/Yosemite/PointOfSale/POSProductProvider.swift

```diff
@@ -25,9 +40,38 @@ public final class POSProductProvider: POSItemProvider {
         return resultsController
     }()
 
+    public func providePointOfSaleItemsFromNetwork() async throws -> [POSItem] {
+        do {
+            let products = try await productsRemote.loadAllSimpleProductsForPointOfSale(for: siteID)
+            return mapProductsToPOSItems(products: products)
+        }
+    }
+
+    // Maps result to POSProduct, and populate the output with:
+    // - Formatted price based on store's currency settings.
+    // - Product categories, if any.
+    // - Product thumbnail, if any.
+    private func mapProductsToPOSItems(products: [Product]) -> [POSItem] {
+        let currencyFormatter = CurrencyFormatter(currencySettings: currencySettings)
+        return products.map { product in
+            let formattedPrice = currencyFormatter.formatAmount(product.price, with: currencySettings.currencyCode.rawValue) ?? "-"
+            let thumbnailSource = product.images.first?.src
+            let productCategories = product.categories.map { $0.name }
+
+            return POSProduct(itemID: UUID(),
```

Comment:
In this case I think is not enough with `productID`, since for example the `CartView` might render the same product multiple times, which causes issues to SwiftUI when cannot identify the elements uniquely only by `productID` and needs other unique identifier, eg:
```
ForEach(viewModel.itemsInCart, id: \.id)
```

I think creating a new UUID should be fine (for now), but I'll check if we may want to use some sort combination for a more stable unique ID.